rules/noUnnecessaryElseRule.ts

```diff
@@ -1,22 +1,56 @@
-import * as ts from 'typescript';
 import * as Lint from 'tslint';
 import * as utils from 'tsutils';
-import { AbstractIfStatementWalker } from '../src/walker';
+import * as ts from 'typescript';
 import { isElseIf } from '../src/utils';
 
 const FAIL_MESSAGE = `unnecessary else`;
+const FAIL_MESSAGE_BLOCK = `unnecessary else block`;
+
+const elseBraceRegex = /^[ \t]*else[ \t]\{(.*)\}[ \t]*$/;
```

Comment:
consider not using regular expressions. you already have an AST, so you don't need to work with the raw source text. This regex for example doesn't handle comments or line breaks